### PR TITLE
link to the failed module - even if no needles were found

### PR DESCRIPTION
**Pull request by coolo**

### CODE REVIEW

**Comment by coolo** [comment]
the datastructure is actually pretty stupid now, but I wanted a minimal change

File: lib/OpenQA/Schema/Result/Jobs.pm [hunk 1]
```diff
@@ -592,6 +593,9 @@ sub failed_modules_with_needles {
             for my $needle (@{$detail->{needles}}) {
                 push( @needles, [ $needle->{name}, $counter ] );
             }
+            if (!@{$detail->{needles}}) {
+                push(@needles, [ undef, $counter ] );
```
Comment (aaannz) [inline lib/OpenQA/Schema/Result/Jobs.pm]: Nitpick: missing space after opening (
I was wondering if we should not add perltidy rule to add spaces after function call (.

Comment (coolo) [inline lib/OpenQA/Schema/Result/Jobs.pm]: that might be your personal preference, but it looks to me this is not openQA perl style:

<pre>
coolo@gertrude#openQA>grep -r 'if.*([^ ]' lib/ | wc -l
405
coolo@gertrude#openQA>grep -r 'if.*( ' lib/ | wc -l
22
</pre>

Comment (coolo) [inline lib/OpenQA/Schema/Result/Jobs.pm]: if you count lib and templates it's even 514 vs 24

Comment (aaannz) [inline lib/OpenQA/Schema/Result/Jobs.pm]: Right, I was referring to the :594, where the space is. And also there is a space before last closing ). I personally don't mind which way we go. You have LGTM and later perltidy rule can be added to consolidate all. I'll create new issue in progress.

Comment (aaannz) [inline lib/OpenQA/Schema/Result/Jobs.pm]: https://progress.opensuse.org/issues/6990

Comment (coolo) [inline lib/OpenQA/Schema/Result/Jobs.pm]: I see now the push - but we're seriously inconsistent with a lot of perl habits. Also the $job->{id} vs $job->{'id'} is killing me :)

Comment (aaannz) [inline lib/OpenQA/Schema/Result/Jobs.pm]: Hehe, I noticed. At SLEPOS we used hashref->{'xx'}. But I try to keep your needs at mind. :)

**Comment by coolo** [comment]
() is just noise for push anyway:

-                push(@needles, [ undef, $counter ] );
+                push @needles, [ undef, $counter ];

**Comment by coolo** [comment]
see 2dc58b750f19f238c5cf44e838d74f721aba58b8 - I pushed this now

File: lib/OpenQA/Schema/Result/Jobs.pm (hunk 1 above)
Comment (coolo) [inline lib/OpenQA/Schema/Result/Jobs.pm]: Vladimir seems to be pretty consistent with {id} while you are consistent with {'id'} :)

**Pull request merged**